test/AutoDiff/leakchecking.swift

```diff
@@ -39,13 +39,13 @@ LeakCheckingTests.testWithLeakChecking("BasicLetLeakChecking") {
   do {
     let model = ExampleLeakModel()
     let x: Tracked<Float> = 1.0
-    _ = model.gradient(at: x) { m, x in m.applied(to: x) }
+    _ = gradient(at: model, x) { m, x in m.applied(to: x) }
   }
 
   do {
     let model = ExampleLeakModel()
     let x: Tracked<Float> = 1.0
-    _ = model.gradient(at: x) { m, x in
+    _ = gradient(at: model, x) { m, x -> Tracked<Float> in
```

Comment:
Note: using the top-level function `gradient(at:_:in:)`, an explicit closure result type `-> Tracked<Float>` becomes necessary. Without it, there's an error:

```
swift/test/AutoDiff/leakchecking.swift:49:22: error: expression type '(_, _)' is ambiguous without more context
      let (y0, y1) = (m.applied(to: x), m.applied(to: x))
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Without the explicit closure result type, there are also other errors, like:
```
swift/test/AutoDiff/leakchecking.swift:48:32: error: unable to infer complex closure return type; add explicit type to disambiguate
    _ = gradient(at: model, x) { m, x in
                               ^
```

These aren't regressions. It's because the old code used the method-style `Differentiable.gradient` function, which was overloaded specifically for `@differentiable (...) -> Tracked<Float>` functions.